src/GogeToken.sol

```diff
@@ -1036,88 +1031,76 @@ contract DogeGaySon is ERC20, Ownable {
             automatedMarketMakerPairs[from] &&
             !excludedAccount
         ) {
-            require(!bots[from] && !bots[to], "bots cannot trade");
-            
-            if (block.timestamp <= _firstBlock.add(_botBlocks)) {
-                bots[to] = true;
-                uint256 toBurn = amount.mul(_botFees).div(100);
-                amount = amount.sub(toBurn);
-                super._transfer(from, deadAddress, toBurn);
-            }
-
+            // if receiver or sender is blacklisted, revert
+            require(!isBlacklisted[from], "GogeToken.sol::_transfer() sender is blacklisted");
+            require(!isBlacklisted[to],   "GogeToken.sol::_transfer() receiver is blacklisted");
             lastReceived[to] = block.timestamp;
-            
-        } else if ( // NON whitelisted sell
+        }
+        
+        else if ( // NON whitelisted sell
             tradingIsEnabled &&
             automatedMarketMakerPairs[to] &&
             !excludedAccount
         ) {
-            require(!bots[from] && !bots[to], 'bots cannot trade');
-                            
+            // if receiver or sender is blacklisted, revert
+            require(!isBlacklisted[from], "GogeToken.sol::_transfer() sender is blacklisted");
```

Comment:
I think that these blacklist `require` statements could be moved to the top of the function. These are checked in each branch of logic and are intended to prevent blacklisted wallets from transferring tokens so why not block them immediately. My bad if you explained this previously.

Reply:
All good. So reason being is first we want to check if the wallet address (receiver, sender, or msg.sender) is excluded from fees (aka whitelisted which could be an admin wallet) which supersedes the blacklist. So yes we don't want a blacklist wallet to transact, UNLESS it's from or to an admin wallet (excludedAccount).